agent/action/mount_disk.go

```diff
@@ -59,11 +64,23 @@ func (a MountDiskAction) Run(diskCid string) (interface{}, error) {
 
 	mountPoint := a.dirProvider.StoreDir()
 
-	isMountPoint, err := a.mountPoints.IsMountPoint(mountPoint)
+	realPath, _, err := a.devicePathResolver.GetRealDevicePath(diskSettings)
+	if err != nil {
+		return nil, bosherr.WrapError(err, "Getting real device path")
+	}
+
+	devicePath, isMountPoint, err := a.diskMounter.IsMountPoint(mountPoint)
 	if err != nil {
 		return nil, bosherr.WrapError(err, "Checking mount point")
 	}
+	a.logger.Info(logTag, "realPath = %s, devicePath = %s, isMountPoint = %s", realPath, devicePath, isMountPoint)
+
 	if isMountPoint {
+		if realPath == devicePath || fmt.Sprint(realPath+"1") == devicePath {
```

Comment:
in our case, we have a fixed persistent disk, and it always be /dev/sdb. And bosh agent always mounts /dev/sdb1 to /var/vcap/store for the first time. when we want to remount, we want to know sdb is already mounted by that line.